wormhole-connect/src/hooks/useAmountValidation.ts

```diff
@@ -0,0 +1,109 @@
+import { amount as sdkAmount, routes } from '@wormhole-foundation/sdk';
+import { useMemo } from 'react';
+import { useSelector } from 'react-redux';
+import { QuoteResult } from 'routes/operator';
+import { RootState } from 'store';
+import { RouteState } from 'store/transferInput';
+
+type HookReturn = {
+  error?: string;
+  warning?: string;
+};
+
+type Props = {
+  balance?: string | null;
+  routes: RouteState[];
+  quotesMap: Record<string, QuoteResult | undefined>;
+  tokenSymbol: string;
+  isLoading: boolean;
+};
+
+export const useAmountValidation = (props: Props): HookReturn => {
+  const { amount } = useSelector((state: RootState) => state.transferInput);
+
+  // Min amount available
+  const minAmount = useMemo(
+    () =>
+      Object.values(props.quotesMap).reduce((minAmount, quoteResult) => {
+        if (quoteResult?.success) {
+          return minAmount;
+        }
+
+        const minAmountError = quoteResult?.error as routes.MinAmountError;
+
+        if (!minAmountError?.min) {
+          return minAmount;
+        }
+
+        if (!minAmount) {
+          return minAmountError.min;
+        }
+
+        const minAmountNum = parseFloat(minAmountError.min.amount);
+        const existingMin = parseFloat(minAmount.amount);
```

Comment:
`parseFloat` is valid here and works, but I want to avoid using floats as much as possible in our amount-related code... just to set a precedent. so do you mind changing this to `BigInt(minAmount.amount)` etc?

Reply:
Tunnel visioned this part, I forgot that `amount.Amount` stores pure integer.